File: numbertheory/archived/factorial.py
def factorial_iterative(n):
    """ O(n) """
    result = 1;
    for i in range(2, n + 1):
        result *= i
    return result

def factorial_recursive(n):
    if n == 0:
        return 1
    return n * factorial_recursive(n - 1)

print(factorial_recursive(42))
print(factorial_iterative(42))

File: numbertheory/archived/factorial.hs
-- Iterative Solution
factorialIterative :: Integer -> Integer
factorialIterative n = product [1..n]

-- Recursive solution
factorialRecursive :: Integer -> Integer
factorialRecursive 0 = 1
factorialRecursive n = n * factorialRecursive(n-1)

main = do
    print (factorialIterative 42)
    print (factorialRecursive 42)